packages/lesswrong/components/form-components/SocialPreviewUpload.tsx

```diff
@@ -0,0 +1,160 @@
+  const originalContents = document.contents?.originalContents
+
+  if (!originalContents) return {description: null, fallbackImageUrl: null}
+  if (originalContents.type !== "html" && originalContents.type !== "ckEditorMarkup") {
+    return {
+      description: `<Description preview not supported for this editor type (${originalContents.type}), switch to HTML or EA Forum Docs [Beta] to see the description preview>`,
+      fallbackImageUrl: null,
+    };
+  }
+
+  const html = dataWithDiscardedSuggestions ?? originalContents.data
+  if (!html) return {description: null, fallbackImageUrl: null}
+  
+  const parser = new DOMParser();
+  const htmlDoc = parser.parseFromString(html, "text/html");
+  // get first img tag
+  const img = htmlDoc.querySelector("img");
+
+  const truncatedHtml = truncate(sanitize(html), PLAINTEXT_HTML_TRUNCATION_LENGTH)
+  const plaintextDescription = htmlToText(truncatedHtml, {
+    wordwrap: false,
+    selectors: [
+      { selector: "img", format: "skip" },
+      { selector: "a", options: { ignoreHref: true } },
+    ],
+  }).substring(0, PLAINTEXT_DESCRIPTION_LENGTH);
+  const previewDesc = getPostDescription({...document, contents: {plaintextDescription}})
+  
+  return {description: previewDesc, fallbackImageUrl: img?.getAttribute("src") || null}
+}
+
+const SocialPreviewUpload = ({name, document, updateCurrentValues, clearField, label, croppingAspectRatio, classes}: {
+  name: string,
+  document: PostsEditWithLocalData,
+  updateCurrentValues: Function,
+  clearField: Function,
+  label: string,
+  croppingAspectRatio?: number,
+  classes: ClassesType
+}) => {
+  const { ImageUpload2 } = Components
+  
+  const urlHostname = new URL(getSiteUrl()).hostname
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const { description, fallbackImageUrl } = useMemo(() => buildPreviewFromDocument(document), [document.contents?.originalContents, document.contents?.dataWithDiscardedSuggestions])
+
+  return (
+    <div className={classes.root}>
+      <div className={classes.preview}>
+        <ImageUpload2
+          name={name}
+          document={document}
+          updateCurrentValues={updateCurrentValues}
+          clearField={clearField}
+          label={"Preview Image"}
+          croppingAspectRatio={croppingAspectRatio}
+          // socialPreviewImageUrl falls back to the first image in the post on save
+          placeholderUrl={fallbackImageUrl || siteImageSetting.get()}
+        />
+        <div className={classes.title}>
+          {document.title}
+        </div>
+        <div className={classes.description}>
+          {description}
+        </div>
+        <div className={classes.url}>
+          {urlHostname}
+        </div>
+      </div>
+      <div className={classes.blurb}>
+        A preview image makes it more likely that people will see your post.
+        <br/><br/>
+        If you're unsure which image to use, consider trying Unsplash or an AI image generator.
```

Comment:
I think it would be useful to link to Unsplash here. Also it looks like the line height needs to be increased.